packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -120,46 +334,53 @@ def reduce_logging():
-    """
-    addrs = [['1.1.1.{}:{}', '1.1.1.{}:{}'],
-             ['/namedvip{}:{}', 'namedvip{}.marathon.l4lb.thisdcos.directory:{}']]
-    samehosts = [True, False]
-    vipnets = ['USER', 'BRIDGE', 'HOST']
-    proxynets = ['USER', 'BRIDGE', 'HOST']
-    permutations = [[vi, va, sh, vn, pn]
-                    for [vi, va] in addrs
-                    for sh in samehosts
-                    for vn in vipnets
-                    for pn in proxynets]
-    tests = [VipTest(i, vi, va, sh, vn, pn) for i, [vi, va, sh, vn, pn] in enumerate(permutations)]
-    threads = []
+    '''Test every permutation of VIP
+    '''
+    # tests
+    # UCR doesn't support BRIDGE mode
+    permutations = [[c, ss, sh, vn, pn]
+                    for c in ['NONE', 'UCR', 'DOCKER']
+                    for ss in [True, False]
+                    for sh in [True, False]
+                    for vn in ['USER', 'BRIDGE', 'HOST']
+                    for pn in ['USER', 'BRIDGE', 'HOST']]
+    tests = [VipTest(i + 1, c, ss, sh, vn, pn) for i, [c, ss, sh, vn, pn] in enumerate(permutations)]
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=maxthreads)
     # deque is thread safe
     failed_tests = deque(tests)
     passed_tests = deque()
     skipped_tests = deque()
+    # skip certain tests
     for r in tests:
-        if not r.samehost and len(dcos_api_session.slaves) == 1:
+        if r.container == 'UCR' or r.container == 'NONE':
+            if r.vipnet == 'BRIDGE' or r.proxynet == 'BRIDGE':
+                r.notes = 'bridge networks are not supported by mesos runtime'
+                failed_tests.remove(r)
+                skipped_tests.append(r)
+                continue
+        if not r.samehost and len(dcos_api_session.all_slaves) == 1:
+            r.notes = 'needs more then 1 agent to run'
             failed_tests.remove(r)
             skipped_tests.append(r)
-            continue
-
-        def run(test):
-            vip_test(dcos_api_session, test)
-            failed_tests.remove(test)
-            passed_tests.append(test)
-
-        threads.append(threading.Thread(target=run, args=([r])))
-    groups = [threads[i:i + maxthreads] for i in range(0, len(threads), maxthreads)]
-    for gs in groups:
-        for t in gs:
-            t.start()
-        for t in gs:
-            t.join()
+    pods = [PodTest(1, ProxyPod.SamePodAsVip, 'HOST', 'UCR', 'HOST', True),
+            PodTest(2, ProxyPod.SamePodAsVip, 'HOST', 'UCR', 'HOST', False)]
+    failed_tests = failed_tests + deque(pods)
+
+    def run(test):
+        test.run(dcos_api_session)
+        failed_tests.remove(test)
+        passed_tests.append(test)
+
+    tasks = [executor.submit(run, t) for t in failed_tests]
+    for t in concurrent.futures.as_completed(tasks):
+        try:
+            t.result()
+        except Exception as exc:
+            log.info('vip_test generated an exception: {}'.format(exc))
     [r.log('PASSED', lvl=logging.INFO) for r in passed_tests]
```

Comment:
Porting from other PR. Once this test is stable, these log lines are not required. You could kill them.